test/testing.sh

```diff
@@ -577,7 +577,6 @@ fail_config_system_config() { rm -f "$testing_lrprefix/etc/luarocks/config.lua";
 test_config_system_config() { mkdir -p "$testing_lrprefix/etc/luarocks"; touch "$testing_lrprefix/etc/luarocks/config.lua"; $luarocks config --system-config; err=$?; rm -f "$testing_lrprefix/etc/luarocks/config.lua"; return $err; }
 fail_config_system_config_invalid() { mkdir -p "$testing_lrprefix/etc/luarocks"; echo "if if if" > "$testing_lrprefix/etc/luarocks/config.lua"; $luarocks config --system-config; err=$?; rm -f "$testing_lrprefix/etc/luarocks/config.lua"; return $err; }
 test_config_user_config() { $luarocks config --user-config; }
-fail_config_user_config() { LUAROCKS_CONFIG="/missing_file.lua" $luarocks config --user-config; }
```

Comment:
I think there was a discussion about this, but I don't remember where. But the situation right now is that this test is not working because the behavior of config loading in luarocks.cfg was changed a while ago (when @Tieske made it smarter). 

The end result is that now, a missing file in $LUAROCKS_CONFIG only causes a warning, and then luarocks.cfg proceeds trying to load the defaults. On my machine, where I do have an existing ~/.luarocks, this ends up returning success, and the test fails. 

I assume that on Travis, where ~/.luarocks doesn't exist, this fails with a file-not-found status, so it gets the expected fail status, but for the wrong reason. In any case, the test, written as is, is misleading.

I deleted it here because I stumbled upon this (again) when running the test suite locally to check these changes. (My approach in this patch was "fix things until the testsuite and luacheck-in-dit seem happy.)